sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxClientCollectionCache.java

```diff
@@ -94,7 +94,7 @@ public RxClientCollectionCache(ISessionContainer sessionContainer,
             retryPolicyInstance.onBeforeSendRequest(request);
         }
 
-        Flux<RxDocumentServiceResponse> responseObs = this.storeModel.processMessage(request);
+        Mono<RxDocumentServiceResponse> responseObs = this.storeModel.processMessage(request);
         return responseObs.map(response -> BridgeInternal.toResourceResponse(response, DocumentCollection.class)
                 .getResource()).single();
```

Comment:
Please remove unnecessary `.single()`

Reply:
as discussed offline. they are intentionally kept as they will do correctness validation to ensure Mono is not empty.